packages/carequality-sdk/src/client/carequality.ts

```
@@ -0,0 +1,123 @@
+  }
+
+  /**
+   * Lists the indicated number of organizations. Mostly used for testing purposes.
+   * @param count number of organizations to fetch
+   * @returns
+   */
+  async listOrganizations(count: number): Promise<OrganizationList> {
+    if (count < 1 || count > 1000)
+      throw new BadRequestError(
+        "Count value must be between 1 and 1000. If you need more, use listDirectories()"
+      );
+    const resp = await this.sendGetRequest(
+      `${Carequality.ORG_ENDPOINT}?apikey=${this.apiKey}&${JSON_FORMAT}&_count=${count}`,
+      { "Content-Type": "application/json" }
+    );
+
+    const bundle: Stu3Bundle = bundleSchema.parse(resp.data.Bundle);
+    const orgs = bundle.entry.map(e => e.resource.Organization);
+
+    return {
+      count: bundle.total.value,
+      organizations: orgs,
+    };
+  }
+
+  /**
+   * Lists the whole Carequality directory.
+   *
+   * @returns organizations list and count
+   */
+  async listAllOrganizations(): Promise<OrganizationList> {
+    let start = 0;
+    const baseUrl = `${Carequality.ORG_ENDPOINT}?apikey=${this.apiKey}&${JSON_FORMAT}&_count=${MAX_COUNT}`;
+    const results: OrganizationList = {
+      count: 0,
+      organizations: [],
+    };
+    let isDone = false;
+
+    while (!isDone) {
+      const url = baseUrl + `&_start=${start}`;
+      try {
+        const resp = await this.sendGetRequest(url);
+        const bundle = bundleSchema.parse(resp.data.Bundle);
+        const organs = bundle.entry.map(e => e.resource.Organization);
+        results.organizations.push(...organs);
+        results.count += bundle.total.value;
+        start += MAX_COUNT;
+
+        const bundleEntryLength = resp.data.Bundle.entry.length;
+        if (bundleEntryLength <= MAX_COUNT / 2) {
+          console.log("Reached the end of the CQ directory...");
+          isDone = true;
+        }
+      } catch (error) {
+        console.log(`Failed to list CQ organizations. Cause: ${error}.`);
+        isDone = true;
```

Maybe we should add a param `failSilently` with default `false` so the client can choose whether to have this behavior? Hiding errors by default is dangerous.